crates/transcode/src/lib.rs

```diff
@@ -739,4 +760,16 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    #[should_panic(expected = "input length was longer than expected by 1 byte(s)")]
+    fn fail_decode_input_with_extra_bytes() {
+        let metadata = generate_metadata();
+        let transcoder = ContractMessageTranscoder::new(metadata);
+
+        let encoded_bytes = hex::decode("633aa55100").unwrap();
+        let _ = transcoder
+            .decode_contract_message(&mut &encoded_bytes[..])
```

Comment:
*nit* this does only test one of the code paths which has the length check.
So could add unit tests for those other 2 `decode_` functions.

Or what would be really nice would be to update the existing integration tests to test these cases: https://github.com/paritytech/cargo-contract/blob/master/crates/cargo-contract/tests/decode.rs#L27